testsuite/model/src/test/java/org/keycloak/testsuite/model/session/OfflineSessionPersistenceTest.java

```diff
@@ -164,6 +166,8 @@ public void testPersistenceMultipleNodesClientSessionAtSameNode() throws Interru
     }
 
     @Test(timeout = 90 * 1000)
+    @RequireProvider(UserSessionPersisterProvider.class)
+    @RequireProvider(value = UserSessionProvider.class, only = InfinispanUserSessionProviderFactory.PROVIDER_ID)
```

Comment:
This is weird one. There's something wrong with map-[realms|user-sessions|client-sessions]-mapStorage-f:ROOT:concurrenthashmap.json file where corresponding storages are serialized. There's probably something missing and I guess it has something to do with concurrency. 
Problem can be reproduced by running `mvn test -f testsuite/model/pom.xml -Phot-rod -Dkeycloak.logging.level=debug -Dtest=OfflineSessionPersistenceTest#testPersistenceMultipleNodesClientSessionsAtRandomNode -Dmaven.surefire.debug` with breakpoint at `ConcurrentHashMapStorageProviderFactory.java:234`

```
Caused by: java.lang.RuntimeException: com.fasterxml.jackson.databind.JsonMappingException: Unexpected end-of-input within/between Object entries
 at [Source: (File); line: 12846, column: 26] (through reference chain: java.util.LinkedList[637])
	at org.keycloak.models.map.storage.chm.ConcurrentHashMapStorageProviderFactory.loadMap(ConcurrentHashMapStorageProviderFactory.java:234)
	at org.keycloak.models.map.storage.chm.ConcurrentHashMapStorageProviderFactory.lambda$getStorage$1(ConcurrentHashMapStorageProviderFactory.java:263)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705)
	at org.keycloak.models.map.storage.chm.ConcurrentHashMapStorageProviderFactory.getStorage(ConcurrentHashMapStorageProviderFactory.java:263)
	at org.keycloak.models.map.storage.chm.ConcurrentHashMapStorageProvider.getStorage(ConcurrentHashMapStorageProvider.java:43)
	at org.keycloak.models.map.userSession.MapUserSessionProviderFactory.create(MapUserSessionProviderFactory.java:86)
	at org.keycloak.models.map.userSession.MapUserSessionProviderFactory.create(MapUserSessionProviderFactory.java:43)
	at org.keycloak.services.DefaultKeycloakSession.getProvider(DefaultKeycloakSession.java:316)
	at org.keycloak.services.DefaultKeycloakSession.sessions(DefaultKeycloakSession.java:455)
	at org.keycloak.testsuite.model.session.OfflineSessionPersistenceTest.lambda$createOfflineClientSession$28(OfflineSessionPersistenceTest.java:282)
	at org.keycloak.testsuite.model.KeycloakModelTest.lambda$withRealm$13(KeycloakModelTest.java:598)
	at org.keycloak.testsuite.model.KeycloakModelTest.lambda$inComittedTransaction$11(KeycloakModelTest.java:563)
	at org.keycloak.testsuite.model.KeycloakModelTest.lambda$inComittedTransaction$12(KeycloakModelTest.java:580)
	at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransaction(KeycloakModelUtils.java:239)
	at org.keycloak.testsuite.model.KeycloakModelTest.inComittedTransaction(KeycloakModelTest.java:568)
	at org.keycloak.testsuite.model.KeycloakModelTest.inComittedTransaction(KeycloakModelTest.java:563)
	at org.keycloak.testsuite.model.KeycloakModelTest.withRealm(KeycloakModelTest.java:595)
	at org.keycloak.testsuite.model.session.OfflineSessionPersistenceTest.createOfflineClientSession(OfflineSessionPersistenceTest.java:281)
	at org.keycloak.testsuite.model.session.OfflineSessionPersistenceTest.lambda$testPersistenceMultipleNodesClientSessionsAtRandomNode$19(OfflineSessionPersistenceTest.java:192)
	at org.keycloak.testsuite.model.KeycloakModelTest.lambda$inIndependentFactories$4(KeycloakModelTest.java:373)
	at org.keycloak.testsuite.model.KeycloakModelTest.inIndependentFactory(KeycloakModelTest.java:472)
```

Reply:
Could it be that each simulated Keycloak instance, the CHM stores should point at is own set of files, but at the moment in this case of concurrent tests they all try to read and write the same file? 

Should that be fixed, so that each of them points at their own set of files, or should the test be ignored for the CHM stores?

Reply:
That might be the case but I'm not sure why we see it only in `testPersistenceMultipleNodesClientSessionAtSameNode` and `testPersistenceMultipleNodesClientSessionsAtRandomNode` tests and not in other tests that use `inIndependentFactories`. I'm also curious about @hmlnarik's view on this.

Reply:
> Could it be that each simulated Keycloak instance, the CHM stores should point at is own set of files, but at the moment in this case of concurrent tests they all try to read and write the same file?

I believe this is the case. Would this vanish with synchronizing the `loadMap` / `storeMap` methods on `ConcurrentHashMapStorageProviderFactory.class`?

Reply:
@hmlnarik - yes, the error messages I see above will then vanish. Then the test fails as the number of stored sessions deviates from the actual number of sessions. IMHO testPersistenceMultipleNodesClientSessionsAtRandomNode and testPersistenceMultipleNodesClientSessionAtSameNode would deliver sensible results once they run against a clustered store (either HotRot or Infinispan). WDYT?

Reply:
Thank you, then it confirms that it is indeed the cause.

Because of this, I agree with disabling the test for CHM map storage

Reply:
... looking a bit more closely, HotRod doesn't provide offline session support yet, so Infinispan is the only truly supported multi-node store?

Reply:
Didn't see your previous comment when writing my comment. Thanks @hmlnarik - resolving this.